src/components/fields/MultiSchemaField.js

```diff
@@ -30,7 +30,49 @@ class AnyOfField extends Component {
 
   getMatchingOption(formData, options) {
     for (let i = 0; i < options.length; i++) {
-      if (isValid(options[i], formData)) {
+      const option = options[i];
+
+      // If the schema describes an object then we need to add slightly more
+      // strict matching to the schema, because unless the schema uses the
+      // "requires" keyword, an object will match the schema as long as it
+      // doesn't have matching keys with a conflicting type. To do this we use an
+      // "anyOf" with an array of requires. This augmentation expresses that the
+      // schema should match if any of the keys in the schema are present on the
+      // object and pass validation.
+      if (option.properties) {
+        // Create an "anyOf" schema that requires at least one of the keys in the
+        // "properties" object
+        const requiresAnyOf = {
+          anyOf: Object.keys(option.properties).map(key => ({
+            required: [key],
+          })),
+        };
+
+        let augmentedSchema;
+
+        // If the "anyOf" keyword already exists, wrap the augmentation in an "allOf"
```

Comment:
I'm guessing that once rjsf implements "allOf", we'll need additional logic here -- maybe we should either 1) proactively add it in or just 2) add a FIXME comment?

Reply:
@epicfaace Unless I'm mistaken, this should already work with a schema that contains `allOf`. If the `allOf` key is present, we shallow clone it and append the new subschema.

Reply:
Oh, yeah, you're right -- thanks!